prov/gni/src/gnix_vector.c

```diff
@@ -160,35 +159,24 @@ inline int _gnix_vec_insert_at(gnix_vector_t *vec, gnix_vec_entry_t *entry,
 
 	if (vec->vector[index]) {
 		GNIX_WARN(FI_LOG_EP_CTRL, "Existing element found in "
-			  "__gnix_vec_insert_last\n");
+			  "__gnix_vec_insert_at\n");
 		return -FI_ECANCELED;
 	} else {
 		vec->vector[index] = entry;
 		return FI_SUCCESS;
 	}
 }
 
-inline int _gnix_vec_remove_first(gnix_vector_t *vec)
-{
-	return _gnix_vec_remove_at(vec, 0);
-}
-
-inline int _gnix_vec_remove_last(gnix_vector_t *vec)
+static inline int __gnix_vec_remove_at(gnix_vector_t *vec,
+				       gnix_vec_index_t index)
 {
-	return _gnix_vec_remove_at(vec, vec->attr.cur_size-1);
-}
-
-inline int _gnix_vec_remove_at(gnix_vector_t *vec, gnix_vec_index_t index)
-{
-	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
-
-	if (unlikely(!vec || index >= vec->attr.cur_size)) {
-		GNIX_WARN(FI_LOG_EP_CTRL, "Invalid parameter to "
-			  "__gnix_vec_remove_at\n");
-		return -FI_EINVAL;
-	} else if (unlikely(vec->state == GNIX_VEC_STATE_DEAD)) {
+	if (unlikely(vec->state == GNIX_VEC_STATE_DEAD)) {
```

Comment:
> You probably don't need to check state here. You should be able to assume, and someone else can correct me, that you won't operate on a dead vector intentionally. Users should never operate on structures that they've closed... so shame on them if they do. This would allow you to remove the state checking which would be a small branching improvement.

I'll wait until profiling data is collected to do this.